doc/README.md

```diff
@@ -1,34 +1,49 @@
 [![Documentation Status (develop)](https://readthedocs.org/projects/gpflow/badge/?version=develop)](http://gpflow.readthedocs.io/en/develop/?badge=develop)
 
-We use readthedocs to build the online documentation, and have separate versions for the `master` and `develop` branches: https://gpflow.readthedocs.io/en/master/ (for the latest official release e.g. on PyPI) and https://gpflow.readthedocs.io/en/develop/ (for the latest cutting-edge code available from the develop branch on github).
+We use readthedocs to build the online documentation, and have separate versions for the `master` and `develop` branches:
+https://gpflow.readthedocs.io/en/master/ (for the latest official release e.g. on PyPI) and
+https://gpflow.readthedocs.io/en/develop/ (for the latest cutting-edge code available from the develop branch on github).
 
-## Compiling documentation
+
+## Compile documentation locally
 
 To compile the GPflow documentation locally:
 
-1. Install doc dependencies
-```
-pip install sphinx sphinx_rtd_theme numpydoc nbsphinx sphinx_autodoc_typehints ipython
-```
-2. Install pandoc
-```
-pip install pandoc
-```
-If pandoc does not install via pip, or step 4 does not work, go to pandoc.org/installing.html
-
-3. Change directory to `doc`
-```
-cd doc
-```
-
-4. Compile the documentation as html
-```
-make html
-```
-
-5. Check documentation locally by opening (in a browser) doc/build/html/index.html
+1. Change to this directory (e.g. `cd doc` if you are in the GPflow git repository's base directory)
+
+2. Install doc dependencies
+   ```
+   pip install sphinx sphinx_rtd_theme numpydoc nbsphinx sphinx_autodoc_typehints ipython
+   ```
+
+3. Install pandoc
+   ```
+   pip install pandoc
+   ```
+   If pandoc does not install via pip, or step 5 does not work, go to pandoc.org/installing.html (the PyPI package depends on the external system-wide installation of pandoc executables)
+
+4. Generate auto-generated files
+   * Notebooks (.ipynb): run `make -C source/notebooks -j 4` (here with 4 parallel threads)
+   * API documentation (.rst): run `python source/generate_module_rst.py`
+
+5. Compile the documentation
+   ```
+   make html
+   ```
+
+6. Check documentation locally by opening (in a browser) build/html/index.html
+
+
+## Setup for automatic documentation generation
+
+Upon each merge to the `develop` branch, this repository's [CircleCI configuration](../.circleci/config.yml) runs the `trigger-docs-generation` step
+which triggers a CircleCI build on the [GPflow/docs repository](https://github.com/GPflow/docs).
+This clones the latest GPflow develop branch and compiles all notebooks from jupytext to .ipynb
+(setting the `DOCS` environment variable so that notebook optimisations are run to convergence)
+and runs the `generate_module_rst.py` script as above to generate the .rst files for API documentation.
```

Comment:
The python script `generate_module_rst.py` runs in circleci containersa (and *not* on readthedocs containers) because it requires `GPflow`, `TensorFlow` and `TensorFlow Probability` to be installed. ReadTheDocs containers, responsible for generating the docs using Sphinx, are not big enough to support these libraries.

Reply:
added an extra sentence to capture this explanation - thanks!